src/dtproject/assets/schemas/devcontainers.json

```diff
@@ -2,11 +2,22 @@
     "$schema": "http://json-schema.org/draft-07/schema",
     "$comment": "Schema for /dtproject/devcontainers.yaml file",
     "type": "object",
-    "description": "Dictionary of named devcontainer-compatible configurations",
-    "patternProperties": {
-        ".*": {
-            "type": "object",
-            "description": "DevContainer-compatible configuration. This is whatever you would put inside a '.devcontainer.json' file"
```

Comment:
This was correct, the `devcontainers.yaml` layer does indeed contain a set of named devcontainer configurations. This is what I have in mind:

```JSON
{
    "$schema": "http://json-schema.org/draft-07/schema",
    "$comment": "Schema for /dtproject/devcontainers.yaml file",
    "type": "object",
    "description": "Dictionary of named devcontainer configurations",
    "patternProperties": {
        ".*": {
            "type": "object",
            "description": "DevContainer configuration object.",
            "properties": {
                "configuration": {
                    "type": "object",
                    "description": "Name of the container configuration to use. Must match one of the configurations defined in the `configurations.yaml` layer.",
                    "default": "default"
                },
               <any devcontainer config properties that are not provided by a docker-compose file>
        }
    }
}
```